docs/siem/detections/api/rules-api-create.asciidoc

```diff
@@ -60,145 +65,235 @@ calls, see:
+|==============================================
 
-* `query`: query-based conditions with or without additional filters.
-* `saved_query`: saved search, identified in the `saved_id` field.
-* `machine_learning`: rule based on a {ml} job's anomaly scores.
+===== Optional fields for query and threshold rules
 
-|Yes
+[width="100%",options="header"]
+|==============================================
+|Name |Type |Description
 
+|exceptions_list |Object[] a|Array of exception containers, which define
+exceptions that prevent the rule from generating alerts even when its other
+criteria are met. The object has these fields:
 
-|threat |<<threats-object-create, threat[]>> |Object containing attack 
-information about the type of threat the rule monitors, see
-{ecs-ref}/ecs-threat.html[ECS threat fields]. |No, defaults to an empty array.
+* `id` (string, required): List ID of the exception container.
+* `namespace_type` (string required): Determines whether the exceptions are
+valid in only the rule's {kib} space (`single`) or in all {kib} spaces
+(`agnostic`). 
+* `type` (string, required): The exception type, which must be either
+a detection rule exception (`detection`) or an endpoint exception (`endpoint`).
 
-|references |String[] |String array containing notes about or references to 
-relevant information about the rule. |No, defaults to an empty array.
+BEN - add link later
+
+|filters |Object[] |The {ref}/query-filter-context.html[query and filter 
+context] array used to define the conditions for when alerts are created from 
+events. Defaults to an empty array.
+
+|index |String[] |Indices on which the rule functions. Defaults to the
+Security Solution indices defined on the {kib} Advanced Settings page
+(*Kibana* → *Stack Management* → *Advanced Settings* →
+`securitySolution:defaultIndex`).
+
+|language |String |Determines the query language, which must be
+`kuery` or `lucene`. Defaults to `kuery`.
 
-|version |Integer |The rule's version number. a|No, defaults to `1`.
+|risk_score_mapping |Object[] a|Overrides generated alerts' `risk_score` with
+a value from the source event:
+
+* `field` (string, required): Source event field used to override the default
+`risk_score`. This field must be an integer.
+* `operator` (string, required): Must be `equals`.
+* `value`(string, required): Must be an empty string (`""`).
+
+|severity_mapping |Object[] a|Overrides generated alerts' `severity` with
+values from the source event:
+
+* `field` (string, required): Source event field used to override the default
+`severity`.
+* `operator` (string, required): Must be `equals`.
+* `severity` (string, required): Mapped severity value, must be `low`,
+`medium`, `high`, or `critical`.
+* `value`(string, required): Field value used to determine the `severity`.
+
+|timestamp_override |String |Determines the time field used to query indices.
+When unspecified, rules query the `@timestamp` field. The source field
+must an {es} date data type.
```

Comment:
Similarly there is also the `rule_name_override` field (added in https://github.com/elastic/kibana/pull/70288), which will take a string field and if the source event has a value for that field, will use that as the `signal.rule.name` when creating the detection alert. This is useful for detection rules whose sole purpose is bringing in external alerts into the detection engine, as this allows the user to expose a bit more information front and center on the alerts table.

We use this internally within the `Elastic Endpoint` rule to map the `message` field to `Rule Name`, as this contains details of the alert like `Malware Prevention Alert`.

Reply:
Thanks - I'm adding this without too much background info. But your example is great, and I'll explain the idea when I document the UI - so thanks!